### ISSUE TITLE
Wrong type inferred in COND # - doesn't detect ref.

### ISSUE THREAD

**Issue by Kaszub09** [abaplint/abaplint#3520]
This code works in ABAP 7.50 and `int_ref` is correctly detected as `type ref to i`.
```abap 
  DATA(int) = 2.
  DATA(int_ref) = COND #( WHEN 1 = 2 THEN REF i( int ) ELSE REF i( int ) ). 
  int_ref->* = 3.
```
But on playground it's detected as `type i`.

**Comment by larshp** [comment]
@Kaszub09 try again, latest version
